plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.ts

```diff
@@ -242,6 +267,200 @@ export class GithubEntityProvider implements EntityProvider {
+    this.logger.debug(`Received event from ${params.topic}`);
+    if (params.topic !== TOPIC_REPO_PUSH) {
+      return;
+    }
+
+    if (params.metadata?.['x-github-event'] === 'push') {
+      await this.onRepoPush(params.eventPayload as PushEvent);
+    }
+
+    return;
+  }
+
+  /** {@inheritdoc @backstage/plugin-events-node#EventSubscriber.supportsEventTopics} */
+  supportsEventTopics(): string[] {
+    return [TOPIC_REPO_PUSH];
+  }
+
+  private async onRepoPush(event: PushEvent) {
+    if (!this.connection) {
+      throw new Error('Not initialized');
+    }
+
+    const repoName = event.repository.name;
+    const repoUrl = event.repository.url;
+    this.logger.debug(`handle github:push event for ${repoName} - ${repoUrl}`);
+
+    const branch =
+      this.config.filters?.branch || event.repository.default_branch;
+
+    if (!event.ref.includes(branch)) {
+      this.logger.debug(`skipping push event from ref ${event.ref}`);
+      return;
+    }
+
+    if (this.config.checkRepositoryFiltersForWebhook) {
+      const repository = await this.getRepositoryInfo(repoName);
+
+      if (!repository) {
+        this.logger.debug(
+          `skipping push event from repository ${repoName} because didn't find information in Github`,
+        );
+        return;
+      }
+
+      const matchingTargets = this.matchesFilters([repository]);
+      if (matchingTargets.length === 0) {
+        this.logger.debug(
+          `skipping push event from repository ${repoName} because didn't match provider filters`,
+        );
+        return;
+      }
+    }
+
+    // the commit has information about the files (added,removed,modified)
+    // so we will process the change based in this data
+    // https://docs.github.com/en/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#push
+    const added = this.collectDeferredEntitiesFromCommit(
+      event.repository.url,
+      branch,
+      event.commits,
+      (commit: Commit) => [...commit.added],
+    );
+    const removed = this.collectDeferredEntitiesFromCommit(
+      event.repository.url,
+      branch,
+      event.commits,
+      (commit: Commit) => [...commit.removed],
+    );
+    const modifiedCatalogFiles = this.collectFilesFromCommit(
+      event.commits,
+      (commit: Commit) => [...commit.modified],
+    );
+
+    const limiter = limiterFactory(10);
+
+    let modified = [];
+    let promises: Promise<void>[] = [];
+
+    if (this.catalogApi && this.tokenManager) {
+      modified = modifiedCatalogFiles
+        .map(filePath => `${event.repository.url}/blob/${branch}/${filePath}`)
+        .map(url => {
+          const location = GithubEntityProvider.toLocationSpec(url);
+
+          return locationSpecToLocationEntity({ location });
+        });
+
+      const { token } = await this.tokenManager.getToken();
+
+      promises = modified.map(entity =>
+        limiter(async () =>
+          this.catalogApi!.refreshEntity(stringifyEntityRef(entity), { token }),
```

Comment:
We can use the refresh method on the connection to trigger refreshes for the entities in the case of the github entity provider.
It would look somehting like this: 
```ts
this.connection.refresh({keys: modifiedCatalogFiles.map(filePath=> `url:${event.repository.url}/tree/${branch}/${filePath}`)})
```
Using this would result in 1 update to refresh all the modified files, if they have a matching refresh_key for them instead of calling entity refresh 1 by 1 for every entity

Reply:
Nice! This will help to avoid the new dependencies too, amazing

Reply:
I need to check whether I can adopt the same. When I first looked into connection.refresh I failed a bit to understand how to use it properly 😅